src/DIRAC/Resources/MessageQueue/StompMQConnector.py

```diff
@@ -200,10 +200,14 @@ def connect(self, parameters=None):
 
         for _ in range(10):
             try:
-                # We need to explicitely call disconnect to avoid leaving
-                # threads behind
+                # We need to explicitly call disconnect to avoid leaving threads behind
                 self.connection.disconnect()
-                self.connection.connect(username=user, passcode=password, wait=True)
+
+                # Connect to the broker with a timeout of 30 seconds
+                self.connection.connect(username=user, passcode=password, wait=False)
```

Comment:
But this is exactly what `wait = True` does

https://github.com/jasonrbriggs/stomp.py/blob/0f8fe0f398604ced511613fcdbd1fa78a8d06a0a/stomp/protocol.py#L338-L341

Reply:
It isn't as `wait=True` blocks forever and doesn't have a timeout.